compiler/pipes/filter-only-actually-used.cpp

```diff
@@ -9,6 +9,75 @@
 #include "compiler/data/src-file.h"
 #include "compiler/compiler-core.h"
 #include "compiler/threading/profiler.h"
+#include "compiler/vertex-util.h"
+
+// having a typed callable __invoke(), which is a virtual function with switch-case dispatching,
+// replace body of `case {lambda_class_to_remove.hash()}:` (which is a lambda invoke call)
+// to just `break`; see below, why this is important
+class RemoveLambdaCallFromTypedCallablePass final : public FunctionPassBase {
```

Comment:
When calling a function lambda from a function lambda, the problem is returned:
```php
class Example {
  public function unused_function(): void {
    $tmp_var = "";
    $this->call_function_first(function() use ($tmp_var): void {
      $this->call_function_second(function() use ($tmp_var) {});
    });
  }

  /** @param callable():void $fn */
  public function call_function_first(callable $fn): void {
    $fn();
  }

  /** @param callable():void $fn */
  public function call_function_second(callable $fn): void {
    $fn();
  }
}
```

Reply:
Nice catch! BTW, the bug existed only in multi-threaded mode, because of a race condition. Fixed in a separate commit, please check